i6_models/parts/rasr_fsa.py

```diff
@@ -194,13 +209,93 @@ def append_fsa(a, b):
         return out_fsa
 
 
-class RasrFsaBuilderV2(RasrFsaBuilder):
+class RasrFsaBatchBuilder(ABC):
     """
-    An update of the RasrFsaBuilder that is compatible with the fbw2 op from i6_native_ops.
+    Abstract base class for building an FSA, compatible with the `fbw2` op from `i6_native_ops`.
+
+    The user must overwrite the :funcref:`build_single` method.
+
+    The TDP scale must be explicitly called when running :funcref:`build_single`.
+    For that, :funcref:`apply_tdp_scale` can be used.
     """
 
-    def build_batch(self, seq_tags: Iterable[str]) -> WeightedFsaV2:
-        fsas = list(map(self.build_single, seq_tags))
+    def __init__(self, config_path: str, tdp_scale: float = 1.0):
+        """
+        :param config_path: Path to the RASR config for the FSA builder.
+        :param tdp_scale: Transition penalty to apply for each edge of the resulting FSA.
+            Applied in the :funcref:`build_batch` method.
+        """
+        import librasr
+
+        self.config_path = config_path
+        config = librasr.Configuration()
+        config.set_from_file(self.config_path)
+        self.builder = librasr.AllophoneStateFsaBuilder(config)
+        self.tdp_scale = tdp_scale
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        del state["builder"]
+        return state
+
+    def __setstate__(self, state):
+        import librasr
+
+        self.__dict__.update(state)
+        config = librasr.Configuration()
+        config.set_from_file(self.config_path)
+        self.builder = librasr.AllophoneStateFsaBuilder(config)
```

Comment:
Optional suggestion for improvement: make setting the builder its own method to avoid code duplication with the `__init__` function.
Even more optional: The state handling is the same as in the `RasrFsaBuilder` so maybe this warrants an even more abstract class but this is not that important.